### ISSUE TITLE
Incorrect stencil access for field vectors

### ISSUE THREAD

**Issue by TeranIvy** [stfc/PSyclone#350]
When trying to access a stencil (cross) from a `chi(3)` field, PSyclone is generating the following: 
```
chi_stencil_map => chi_proxy%vspace%get_stencil_dofmap(STENCIL_CROSS,1)
 ```
when it should actually be
```
chi_stencil_map => chi_proxy(1)%vspace%get_stencil_dofmap(STENCIL_CROSS,1)
```
For info, the `chi` field is on `ANY_SPACE_2`  in the below metadata:
```
type, public, extends(kernel_type) :: one2one_w2_filter_stencil_kernel_type
    private
    type(arg_type) :: meta_args(4) = (/                       &
        arg_type(GH_FIELD,   GH_WRITE, W3),                   & 
        arg_type(GH_FIELD,   GH_READ,  W3, STENCIL(CROSS)),   & 
        arg_type(GH_FIELD,   GH_READ,  ANY_SPACE_1, STENCIL(CROSS)),    &
        arg_type(GH_FIELD*3, GH_READ,  ANY_SPACE_2, STENCIL(CROSS))    &
        /)
    type(func_type) :: meta_funcs(1) = (/     &
        func_type(ANY_SPACE_2, GH_DIFF_BASIS) &
        /)
    integer :: iterates_over = CELLS
    integer :: gh_shape = GH_EVALUATOR
  contains
    procedure, nopass :: one2one_w2_filter_stencil_code
  end type
```

**Comment by rupertford** [comment]
Created branch 350_field_vector_stencil_bug

**Comment by rupertford** [comment]
This is a simple fix. I needed to replace `arg.proxy_name` with `arg.proxy_name_indexed` in two places. However, with this change the tests still pass which means we have never checked for this case - which is a bit bad. I will there add in two new tests for the two cases above.

**Comment by TeranIvy** [comment]
Merged #351 so closing this issue.
